main.tf

```diff
@@ -17,12 +17,12 @@ resource "aws_elasticache_replication_group" "redis" {
 }
 
 resource "aws_elasticache_parameter_group" "redis_parameter_group" {
-  name        = "tf-redis-${var.name}-${data.aws_vpc.vpc.tags["Name"]}"
+  name        = "tf-redis-${var.name}-${replace(format("%.100s", lower(replace("${data.aws_vpc.vpc.tags["Name"]}", "_", "-"))), "/\\s/", "-")}"
```

Comment:
looks good, but small improvement:
`name = "${replace(format("%.100s", lower(replace("tf-redis-${var.name}-${data.aws_vpc.vpc.tags["Name"]}", "_", "-"))), "/\\s/", "-")}"`

This way they whole value is guaranteed to be at most 100 chars and equally formatted. It is rather common to use "_" in names, for instance.